app/webpacker/controllers/bulk_actions_controller.js

```diff
@@ -1,24 +1,43 @@
-import ApplicationController from "./application_controller";
-
-export default class extends ApplicationController {
-  static targets = ["extraParams"]
-  static values = { reflex: String }
+import { Controller } from "stimulus";
 
+export default class extends Controller {
   connect() {
-    super.connect();
+    this.element.addEventListener('turbo:submit-end', this.closeModal.bind(this));
+    document.addEventListener('modal-open', this.modalOpen.bind(this));
+    document.addEventListener('modal-close', this.modalClose.bind(this));
   }
 
-  perform() {
-    let params = { bulk_ids: this.getSelectedIds() };
+  disconnect() {
+    this.element.removeEventListener('turbo:submit-end', this.closeModal);
+    document.removeEventListener('modal-open', this.modalOpen);
+    document.removeEventListener('modal-close', this.modalClose);
+  }
 
-    if (this.hasExtraParamsTarget) {
-      Object.assign(params, this.extraFormData())
-    }
+  printInvoices(e) {
+    const csrfToken = document.querySelector('meta[name="csrf-token"]')?.getAttribute('content');
+    const data = { bulk_ids: this.getSelectedIds() };
 
-    this.stimulate(this.reflexValue, params);
+    fetch(e.target.getAttribute('data-url'), {
+      method: "POST",
+      headers: {
+        Accept: "text/vnd.turbo-stream.html",
+        "Content-Type": "application/json",
+        "X-CSRF-Token": csrfToken
+      },
+      body: JSON.stringify(data),
+    })
+      .then((response) => response.text())
+      .then((html) => {
+        Turbo.renderStreamMessage(html);
+      })
+      .catch((error) => console.error(error));
```

Comment:
Maybe I am missing something here, but it feels like we are replicating something that `Turbo` should be doing for us ? Is there a way to refactor this so we don't have to manually use `fetch` ?

Reply:
I see, maybe a comment would be useful here.